portal/app/algorithms/battery_optimizer.py

```diff
@@ -0,0 +1,671 @@
+import json, datetime, numpy as np
+import pandas as pd
+from app.farm_updater import sonnen_api
+from django.conf import settings
+import os
+import cvxpy as cvx
+from statsmodels.tsa.statespace.sarimax import SARIMAX
+from pytz import timezone
+import pytz
```

Comment:
All the dependencies in this list that are not on the `requirements.txt` need to be added. i.e. pandas and whatever statsmodels is

Reply:
forgot about it - done